src/core/scope.c

```diff
@@ -571,6 +571,30 @@ static const char* const scope_result_table[_SCOPE_RESULT_MAX] = {
         [SCOPE_FAILURE_TIMEOUT] = "timeout",
 };
 
+static int scope_freeze(Unit *s) {
```

Comment:
These seem to be exactly the same for scope, service, whatever -- we should just create unit_freeze_common and unit_thaw_common